reporter/src/test/kotlin/utils/FreeMarkerTemplateProcessorTest.kt

```diff
@@ -0,0 +1,187 @@
+    ),
+    analyzer = AnalyzerRun(
+        environment = Environment(),
+        config = DEFAULT_ANALYZER_CONFIGURATION,
+        result = AnalyzerResult.EMPTY.copy(
+            projects = sortedSetOf(
+                Project.EMPTY.copy(
+                    id = Identifier("NPM:@ort:project-in-root-dir:1.0"),
+                    definitionFilePath = "package.json",
+                    vcsProcessed = PROJECT_VCS_INFO
+                ),
+                Project.EMPTY.copy(
+                    id = Identifier("SpdxDocumentFile:@ort:project-in-sub-dir:1.0"),
+                    definitionFilePath = "sub-dir/project.spdx.yml",
+                    vcsProcessed = PROJECT_VCS_INFO
+                ),
+                Project.EMPTY.copy(
+                    id = Identifier("SpdxDocumentFile:@ort:project-in-nested-vcs:1.0"),
+                    definitionFilePath = "nested-vcs-dir/project.spdx.yml",
+                    vcsProcessed = NESTED_VCS_INFO
+                )
+            )
+        )
+    ),
+    scanner = ScannerRun(
+        environment = Environment(),
+        config = ScannerConfiguration(),
+        results = ScanRecord(
+            scanResults = sortedSetOf(
+                scanResultContainer(
+                    id = "NPM:@ort:project-in-root-dir:1.0",
+                    vcsInfo = PROJECT_VCS_INFO,
+                    findingsPaths = listOf(
+                        "src/main.js",
+                        "sub-dir/src/main.cpp",
+                        "nested-vcs-dir/src/main.cpp"
+                    )
+                ),
+                scanResultContainer(
+                    id = "SpdxDocumentFile:@ort:project-in-sub-dir:1.0",
+                    vcsInfo = PROJECT_VCS_INFO,
+                    findingsPaths = listOf(
+                        "sub-dir/src/main.cpp"
+                    )
+                ),
+                scanResultContainer(
+                    id = "SpdxDocumentFile:@ort:project-in-nested-vcs:1.0",
+                    vcsInfo = NESTED_VCS_INFO,
+                    findingsPaths = listOf(
+                        "src/main.cpp"
+                    )
+                )
+            ),
+            storageStats = AccessStatistics()
+        )
+    )
+)
+
+class FreeMarkerTemplateProcessorTest : WordSpec({
+    "deduplicateProjectScanResults" should {
+        val targetProjects = setOf(
+            Identifier("SpdxDocumentFile:@ort:project-in-sub-dir:1.0"),
+            Identifier("SpdxDocumentFile:@ort:project-in-nested-vcs:1.0")
+        )
+
+        val ortResult = ORT_RESULT.deduplicateProjectScanResults(targetProjects)
+
+        "keep the findings of all target projects" {
+            targetProjects.forAll { targetProject ->
+                ortResult.getScanResultsForId(targetProject) shouldBe ORT_RESULT.getScanResultsForId(targetProject)
+            }
+        }
+
+        "filter the findings of the root project as expected" {
```

Comment:
More explicit: "remove the findings of all target projects from the root project"

Reply:
Done.